autograder/core/models/group/group.py

```diff
@@ -93,13 +94,30 @@ def member_names(self) -> List[str]:
     project = models.ForeignKey(Project, related_name="groups",
                                 on_delete=models.CASCADE)
 
-    extended_due_date = models.DateTimeField(
+    @property
+    def extended_due_date(self) -> Optional[datetime]:
```

Comment:
Add docstring:

"""
DEPRECATED. For API legacy use, you can continue to use
this field as usual. Do NOT mix usage of this field and the
two new "hard_extended_due_date" and "soft_extended_due_date" fields.

When this field is set, it indicates that members
of this submission group can submit until this specified
date, overriding the project closing time.
Default value: None
"""